core/platform/auth/firebase_auth_services.py

```diff
@@ -136,6 +128,40 @@ def authenticate_request(request):
     return None if not auth_id else auth_domain.AuthClaims(auth_id, email)
 
 
+def delete_associations(user_id):
+    """Deletes associations referring to the given user_id.
+
+    Args:
+        user_id: str. The ID of the user being deleted.
+
+    Returns:
+        bool. Whether the user's data has been successfully deleted.
+    """
+    assoc = auth_models.UserIdByFirebaseAuthIdModel.get_by_user_id(user_id)
+    if assoc is None:
+        return
+    try:
+        with _acquire_firebase_context():
+            firebase_auth.delete_user(assoc.id)
+    except (ValueError, firebase_exceptions.FirebaseError) as e:
+        logging.exception(e)
+
+
+def are_associations_deleted(user_id):
+    """Returns whether the Firebase account of the given user ID is deleted."""
+    assoc = auth_models.UserIdByFirebaseAuthIdModel.get_by_user_id(user_id)
+    if assoc is None:
+        return True
+    try:
+        with _acquire_firebase_context():
+            firebase_auth.get_user(assoc.id)
+    except firebase_auth.UserNotFoundError:
+        return True
+    except (ValueError, firebase_exceptions.FirebaseError) as e:
+        logging.exception(e)
```

Comment:
Interesting, do you think we should be using this everywhere rather than logging.error? /cc @vojtechjelinek

Reply:
I think in some places it might make sense to use this.

Reply:
Yup, according to docs this is a wrapper around logging.error that additionally adds a stack trace: https://docs.python.org/2/library/logging.html#logging.exception. It only works in `except` blocks.